app/controllers/concerns/basic_authenticable.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+# Enforces HTTP Basic Auth
+module BasicAuthenticable
+  class << self
+    def required?
+      Settings.basic_auth.enabled
+    end
+
+    def authenticate(username, password)
+      validate(username, password)
+    end
+
+    def validate(username, password)
+      utils.secure_compare(::Digest::SHA256.hexdigest(auth_username), ::Digest::SHA256.hexdigest(username)) &
+        utils.secure_compare(::Digest::SHA256.hexdigest(auth_password), ::Digest::SHA256.hexdigest(password))
+    end
+
+    def auth_username
+      @auth_username ||= Settings.basic_auth.username
```

Comment:
Needs to be added to QA settings before we can test this PR out.

Reply:
@avinhurry I can show you how to add the secrets to PasS if you need a hand 👍